.github/workflows/release-next-rspack.yml

```
@@ -0,0 +1,123 @@
+name: Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      dry-run:
+        description: 'Run in dry-run mode (no actual publishing)'
+        required: false
+        default: false
+        type: boolean
+      npm-tag:
+        description: 'NPM tag for publishing'
+        required: false
+        default: 'latest'
+        type: choice
+        options:
+          - latest
+          - alpha
+          - beta
+          - canary
+
+env:
+  DEBUG: napi:*
+
+jobs:
+  build:
+    name: Build
+    uses: rspack-contrib/rspack-toolchain/.github/workflows/build.yml@v1
+    with:
+      package-json-path: rspack/crates/binding/package.json
+      napi-build-command: pnpm build --release
+
+  release:
+    runs-on: ubuntu-latest
+    environment: npm
+    name: Release
+    permissions:
+      contents: write
+      id-token: write
+    needs: [build]
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Display release mode
+        run: |
+          echo "🚀 Release Configuration:"
+          echo "  - Dry-run mode: ${{ inputs.dry-run }}"
+          echo "  - NPM tag: ${{ inputs.npm-tag || 'latest' }}"
+          if [ "${{ inputs.dry-run }}" == "true" ]; then
+            echo "  - ⚠️  This is a DRY RUN - no packages will be published"
+          else
+            echo "  - 📦 This will PUBLISH packages to npm"
+          fi
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '22'
+
+      - name: Enable corepack
+        run: corepack enable
+
+      - name: Setup pnpm
+        run: corepack prepare
+
+      - name: Cache pnpm dependencies
+        uses: actions/cache@v3
+        with:
+          path: ~/.pnpm-store
+          key: ${{ runner.os }}-pnpm-${{ hashFiles('**/pnpm-lock.yaml') }}
+          restore-keys: |
+            ${{ runner.os }}-pnpm-
```

Maybe also add arch here, just to be safe: https://docs.github.com/en/actions/reference/workflows-and-actions/contexts#runner-context

```suggestion
          key: ${{ runner.os }}-${{ runner.arch }}-pnpm-${{ hashFiles('**/pnpm-lock.yaml') }}
          restore-keys: |
            ${{ runner.os }}-${{ runner.arch }}-pnpm-
```